client/qiskit_serverless/core/files.py

```diff
@@ -120,7 +120,7 @@ def delete(self, file: str, provider: Optional[str] = None) -> Optional[str]:
         """Deletes file uploaded or produced by the programs,"""
         tracer = trace.get_tracer("client.tracer")
         with tracer.start_as_current_span("files.delete"):
-            response_data = safe_json_request(
+            response_data = safe_json_request_as_dict(
```

Comment:
I think one day we will need to remove this and just use requests and do proper error handling with correct excetions, etc. 

Definitely not the highest priority. Maybe we can create good first issue or something to handle this in the future.

Reply:
Yeah, i just tried to fix some types. There are too many `Any`s everywhere.

Reply:
Agree!